linters.go

```diff
@@ -53,27 +53,29 @@ var predefinedPatterns = map[string]string{
 	"PATH:LINE:MESSAGE":     `^(?P<path>.*?\.go):(?P<line>\d+):\s*(?P<message>.*)$`,
 }
 
-func getLinterByName(name string, customSpec string) *Linter {
-	if customSpec != "" {
-		return parseLinterSpec(name, customSpec)
+func getLinterByName(name string, overrideConf LinterConfig) *Linter {
+	conf := defaultLinters[name]
+	if val := overrideConf.Command; val != "" {
+		conf.Command = val
 	}
-	linter, _ := NewLinter(defaultLinters[name])
+	if val := overrideConf.Pattern; val != "" {
+		conf.Pattern = val
+	}
+	linter, _ := NewLinter(conf)
 	return linter
 }
 
-func parseLinterSpec(name string, spec string) *Linter {
+func parseLinterSpec(name string, spec string) (LinterConfig, error) {
```

Comment:
I guess this should be called `parseLinterConfigSpec` now that it returns a `LinterConfig`